src/shared/parameter_v2/config/examples/example.yaml

```diff
@@ -0,0 +1,140 @@
+# This file provides an example for how to define a config file, including examples
+# of all parameter types, special attributes, and how to include other configs
+#
+# Parameter schema
+#
+# type:
+#   name:
+#   value:
+#   description:
+#   other attributes...
+#
+# Notes:
+# - All 'name' attributes must be unique per-file
+# - All 'name' attribute values should be snake_case. Spaces are not allowed.
+# - All config file names must be unique
+
+#########################################################################################
+# The following are examples of how to include other configs
+#########################################################################################
+# Config files can include other configs. This is how the parameter hierarchy is built
+# up, and allows us to share parameters between configs. For example, this makes it
+# easy to define some passing parameters in one location, and share those parameters
+# with the subset of Plays that need them.
+#
+# All included config files need to be listed under a single `include` section.
+# Because all config filenames must be unique, just the filename needs to be
+# specified (not the path).
+#
+# In the generated code, included configs and their attributes can be accessed by
+# the name of the included config.
+# Eg. foo.foo_bool
+include:
+  - "foo.yaml"
+---
```

Comment:
Ok so in order for the YAML to be parsed successfully, we need to "terminate" this `include` list somehow. This is because you can't have a "top-level list" with a dictionary list like this.

The two options are:
* terminate the list with `---` (what we have now)
* Put all parameters under their own dictionary

Eg.
```
include:
  - "foo.yaml"
parameters:
  - bool:
      name: example_bool_param
      value: true
      description: >
        Can be true or false
  - int:
      name: example_bool_param
      min: 0
      max: 5
      value: 3
      description: >
        Can be any integer value in the range [min, max]
```

Do people have a preference for one or the other?

Reply:
@jonathanlew I don't know if :+1: means you like the first or second option

Reply:
I mean to say that's a good point you bring up. I personally have no strong feelings either way

Reply:
I vouch for `---` obvious separation from includes to rest of the schema